File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.py
def solution(n, stations, w):
    answer = 0
    
    cnt = 2*w+1
    start = 1
    
    for station in stations:
        if (station - w - start) > 0:
            answer += (station - w - start) // cnt
            if (station - w - start) % cnt:
                answer += 1
        start = station + w + 1
    if n - start + 1 > 0:
        answer += (n-start+1)//cnt
        if (n-start+1)%cnt:
            answer+=1
            
        

    return answer

File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.java
import java.util.*;

class Solution {
    public int solution(int n, int[] stations, int w) {
        int answer = 0;
        
        int cover = 2 * w + 1;
        int start = 1;
        for (int station : stations) {
            int left = station - w;
            
            if (start < left) {
                int gap = left - start;
                answer += gap / cover;
                if (gap % cover != 0) answer++;
            }
            
            start = station + w + 1;
        }
        
        if (start <= n) {
            int gap = n - start + 1;
            answer += gap / cover;
            if (gap % cover != 0) answer++;
        }
 

        return answer;
    }
}